specification/developerhub/resource-manager/Microsoft.DevHub/preview/2022-10-11-preview/workflow.json

```diff
@@ -0,0 +1,1008 @@
+        }
+      },
+      "description": "The response from List Workflows operation."
+    },
+    "PullRequestStatus": {
+      "type": "string",
+      "readOnly": true,
+      "enum": [
+        "unknown",
+        "submitted",
+        "merged",
+        "removed"
+      ],
+      "x-ms-enum": {
+        "name": "PullRequestStatus",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "unknown",
+            "description": "Pull Request state unknown."
+          },
+          {
+            "value": "submitted",
+            "description": "Pull Request submitted to repository."
+          },
+          {
+            "value": "merged",
+            "description": "Pull Request merged into repository."
+          },
+          {
+            "value": "merged",
+            "description": "Workflow no longer found within repository."
+          }
+        ]
+      },
+      "description": "The status of the Pull Request submitted against the users repository."
+    },
+    "DeploymentProperties": {
+      "type": "object",
+      "properties": {
+        "manifestType": {
+          "$ref": "#/definitions/ManifestType"
+        },
+        "kubeManifestLocations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "helmChartPath": {
+          "type": "string",
+          "description": "Helm chart directory path in repository."
+        },
+        "helmValues": {
+          "type": "string",
+          "description": "Helm Values.yaml file location in repository."
+        },
+        "overrides": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "title": "Manifest override values.",
+          "description": "Manifest override values."
+        }
+      }
+    },
+    "ArtifactGenerationProperties": {
+      "type": "object",
+      "properties": {
+        "generationLanguage": {
+          "$ref": "#/definitions/GenerationLanguage"
+        },
+        "languageVersion": {
+          "type": "string",
+          "description": "The version of the language image used for execution in the generated dockerfile."
+        },
+        "builderVersion": {
+          "type": "string",
+          "description": "The version of the language image used for building the code in the generated dockerfile."
+        },
+        "port": {
+          "type": "string",
+          "description": "The port the application is exposed on."
+        },
+        "appName": {
+          "type": "string",
+          "description": "The name of the app."
+        },
+        "enableDockerfileGeneration": {
+          "type": "boolean",
+          "description": "Whether to enable generation of Dockerfile."
+        },
+        "enableManifestGeneration": {
+          "type": "boolean",
+          "description": "Whether to enable generation of deployment and ingress manifests."
+        },
```

Comment:
consider updating to boolean : 

[https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
thank you this is a good point. I've integrated this toggle into the ManifestType enum as "none" and changed the dockerfile generation boolean to an enum for the `enabled` and `disabled` modes